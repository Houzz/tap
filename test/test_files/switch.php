<?php

/**
 * @param int|string $x
 */
function f($x) {
    switch ($x) {
    case 1:
        $y = 1;
        $z = 1;
        break;
    case "str":
        $y = "str";
        $z = "str";
        break;
    case true:  // error
        $z = 1;
        break;
    default:
        $z = 2;
    }
    __tap_check($y, "int|string|null");
    __tap_check($z, "int|string");
}
