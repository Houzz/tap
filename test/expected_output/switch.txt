16 E_SWITCH_CASE_TYPE
22 E_NAME_UNDEFINED
